app/routes/goal_routes.py

```diff
@@ -0,0 +1,96 @@
+from app import db
+from app.models.goal import Goal
+from app.models.task import Task
+from flask import Blueprint, request, jsonify, make_response
+from app.routes.task_routes import validate_id
+
+goal_bp = Blueprint("goals", __name__, url_prefix="/goals")
+
+# ================================
+# Create one goal 
+# ================================
+
+@goal_bp.route("", methods=["POST"])
+def create_goal():
+    request_body = request.get_json()
+
+    try:
+        new_goal = Goal(title=request_body["title"])
+    except KeyError:
+        return make_response({"details": "Invalid data"}, 400)
+
+    db.session.add(new_goal)
+    db.session.commit()
+
+    return jsonify({"goal": new_goal.to_json()}), 201
+
+# ==================================
+# Get all goals  
+# ==================================
+
+@goal_bp.route("", methods=["GET"])
+def get_all_goals():
+        all_goals = Goal.query.all()
+        results_list = []
+        for goal in all_goals:
+            results_list.append(goal.to_json())
+        return jsonify(results_list), 200
```

Comment:
Here, though, you _do_ have to use `jsonify` since `results_list` is a list, not a dictionary.

That being said, if your HTTP status code is 200, you don't _need_ to include it in your return value, so you could just do `return jsonify(results_list)`.

That being said, keeping it can be easily justified as a way to be more explicit, which is often good style IMO.